compiler/src/org.graalvm.compiler.lir.amd64/src/org/graalvm/compiler/lir/amd64/vector/AMD64Packing.java

```diff
@@ -190,27 +246,32 @@ private static void const2addr(CompilationResultBuilder crb, AMD64MacroAssembler
         }
     }
 
-    private static void move(CompilationResultBuilder crb, AMD64MacroAssembler masm, AMD64Address dst, Value src, AMD64Kind srcKind, Register scratch) {
-        if (isRegister(src)) {
-            reg2addr(crb, masm, srcKind, dst, asRegister(src));
-        } else if (isStackSlot(src)) {
-            addr2reg(crb, masm, srcKind, scratch, (AMD64Address) crb.asAddress(src));
-            reg2addr(crb, masm, srcKind, dst, scratch);
-        } else if (isJavaConstant(src)) {
-            const2addr(crb, masm, srcKind, dst, asJavaConstant(src));
-        }
-    }
-
+    /**
+     * This operation loads a vector register with contiguous values from an array. It does so with
+     * the aid of some temporary stack space to minimize expensive vector instructions.
+     */
     public static final class LoadStackOp extends AMD64LIRInstruction {
-
-        private static final int YMM_LENGTH_IN_BYTES = 32;
         public static final LIRInstructionClass<LoadStackOp> TYPE = LIRInstructionClass.create(LoadStackOp.class);
 
+        // Our destination should always be a register.
         @Def({REG}) private AllocatableValue result;
+
+        // We need a scratch register for any possible memory-to-memory moves (see move(...) above).
```

Comment:
I think there's some way to link to other methods? Might be worth investigating the exact Javadoc syntax for this but they use it a bunch throughout the codebase